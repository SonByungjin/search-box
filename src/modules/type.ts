export interface CommonError {
  code: number;
  message: string;
  data?: string;
}
