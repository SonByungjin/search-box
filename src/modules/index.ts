import { combineReducers } from "redux";
import { all, call } from "redux-saga/effects";

import { ItemListReducer, ItemListState } from "./itemList";

import { DetailReducer, DetailState } from "./detail/reducer";
import { rootDetailSaga } from "./detail/sagas";

import { SearchReducer, SearchState } from "./search";
import { rootSearchSaga } from "./search/sagas";

export interface RootState {
  itemList: ItemListState;
  search: SearchState;
  detail: DetailState;
}

export const rootReducer = combineReducers({
  itemList: ItemListReducer,
  search: SearchReducer,
  detail: DetailReducer,
});

export function* rootSaga() {
  yield all([
    call(rootSearchSaga),
    call(rootDetailSaga),
  ]);
}
