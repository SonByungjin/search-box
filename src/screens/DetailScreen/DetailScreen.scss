.pg_detail {
  &__wrapper {
    width: 100vw;
    height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
    > div {
      max-width: 400px;
      width: 100%;
      margin-bottom: 10px;
    }
  }

  &__nav {
    margin-top: 5%;
    > button {
      cursor: pointer;
      font-size: 30px;
    }
  }

  &__count,
  &__uri {
    display: flex;
    align-items: center;
    justify-content: space-between;
    font-size: 30px;
  }
}
