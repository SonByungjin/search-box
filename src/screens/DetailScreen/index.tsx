import { FC } from "react";

const DetailScreen: FC = () => {
  return <div>detail</div>;
};

export { DetailScreen };
