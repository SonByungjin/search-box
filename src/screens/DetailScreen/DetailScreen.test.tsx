import * as React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { createStore } from "redux";
import renderer from "react-test-renderer";

import { rootReducer } from "../../modules";
import { DetailScreen } from "../DetailScreen";

const store = createStore(rootReducer);

describe("SearchScreen", () => {
  test("show render", () => {
    const view = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <DetailScreen />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(view).toMatchSnapshot();
  });
});
