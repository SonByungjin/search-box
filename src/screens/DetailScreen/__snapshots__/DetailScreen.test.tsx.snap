// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailScreen check Snapshot 1`] = `
<div
  className="pg_detail__wrapper"
>
  <div
    className="pg_detail__nav"
  >
    <button
      onClick={[Function]}
    >
      뒤로 가기
    </button>
  </div>
  <div
    className="pg_detail__count"
  >
    <span>
      조회수:
    </span>
    <span>
      0
    </span>
  </div>
  <div
    className="pg_detail__uri"
  >
    <span>
      URI:
    </span>
    <span>
      .undefined
    </span>
  </div>
</div>
`;
